### ISSUE TITLE
Stitching feature request: Transform non-geospatial model inference back into geospatial space. 

### ISSUE THREAD

**Issue by EricKeenan** [microsoft/torchgeo#1440]
### Summary

`GridGeoSampler` allows for efficient imagery chipping in preparation for ML-based inference. My problem is converting these non-geospatial inference results back into geospatial space. E.g. an inference file in `geojson` format that contains model inference bounding boxes, segmentations , etc. 

Does this functionality already exist in `TorchGeo`? If not, how can we make this happen?

### Rationale

From my perspective, this would enable machine learning experts to more easily interact with geospatial data.

### Implementation

I have not thought deeply about implementation.

### Alternatives

Not that I am aware of. Although there do seem to be related issues and PRs, e.g. https://github.com/microsoft/torchgeo/issues/1407.

### Additional information

I would be interested in contributing to a solution. However, I'm not quite sure how to get started.

**Comment by adamjstewart** [comment]
Yes, #1407 is likely going to be the limiting factor here. I think the first step is to figure out how to keep things like CRS and transform in the sample batch. Maybe @adriantre has additional thoughts.
